src/transformStateWithClones.js

```diff
@@ -7,7 +7,48 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const clone = { ...state };
+  const stateHistory = [];
+
+  for (const action of actions) {
```

Comment:
[CODE STYLE] - don't mutate object or arrays - it will cause unexpected results later on. You should make a copy of the state inside the loop for each action to ensure that each state in the history is independent.